sw/device/lib/crypto/impl/mac.c

```diff
@@ -219,56 +172,81 @@ crypto_status_t otcrypto_hmac_init(hmac_context_t *ctx,
   uint32_t *share1;
   HARDENED_TRY(keyblob_to_shares(key, &share0, &share1));
 
-  // TODO: Once we have hardened SHA-256, do not unmask the key here.
+  // TODO: Once we have hardened SHA2, do not unmask the key here.
   uint32_t unmasked_key[keyblob_share_num_words(key->config)];
   for (size_t i = 0; i < keyblob_share_num_words(key->config); i++) {
     unmasked_key[i] = share0[i] ^ share1[i];
   }
 
   // Initialize the key block, K0. See FIPS 198-1, section 4.
-  uint32_t k0[kSha256MessageBlockWords] = {0};
-  if (key->config.key_length <= kSha256MessageBlockBytes) {
-    // If the key fits into the SHA-256 block size, we just need to copy it
+  uint32_t k0[message_block_words];
+  memset(k0, 0, sizeof(k0));
+  if (key->config.key_length < (digest_words * sizeof(uint32_t))) {
+    // Key is too short.
+    return OTCRYPTO_BAD_ARGS;
```

Comment:
Forgot to mention it, but this bit resolves https://github.com/lowRISC/opentitan/issues/19321